pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -53,9 +54,10 @@ type DataSource struct {
 	ProxyOpts         *proxy.Options
 	AWSConfigProvider awsauth.ConfigProvider
 
-	logger          log.Logger
-	tagValueCache   *cache.Cache
-	resourceHandler backend.CallResourceHandler
+	logger                 log.Logger
+	tagValueCache          *cache.Cache
+	resourceHandler        backend.CallResourceHandler
+	monitoringAccountCache sync.Map
```

Comment:
Should I make this a `*cache.Cache` like `tagValueCache` above? That would allow setting a `ttl` for this, although I don't believe this really changes unless the AWS account has cross account observability turned off for it.